File: Array-String/String/Python/replace_all_question_marks_to_avoid_consecutive_repeating_characters.py
class Solution:
    def modifyString(self, s: str) -> str:
        s = list(s)
        n = len(s)
        if n == 1 and s == "?": return "a"
        for i in range(n):
            if s[i] == '?':
                for j in range(97, 123):
                    if (i > 0 and s[i-1] == chr(j)) or (i < n - 1 and s[i+1] == chr(j)): continue
                    s[i] = chr(j)
                    break
        return ''.join(s)

File: Array-String/String/CPP/replace_all_question_marks_to_avoid_consecutive_repeating_characters.cpp
#include <string>

using namespace std;

class Solution {
    public:
        string modifyString(string s) {
            int n = s.size(), i = 0;
            if (n == 1 && s == "?") return "a";
            for (i; i < n; i++) {
                if (s[i] == '?') {
                    int l = i <= 0 ? -1 : s[i-1] - 'a',
                        r = i > n ? -1 : s[i+1] - 'a',
                        cur = 0;
                    while (cur < 26) {
                        if (cur != l && cur != r) break;
                        cur++;
                    }
                    s[i] = (char)cur+97;
                }
            }
            return s;
        }

        // cleaner solution
        string modifyString(string s) {
            int n = s.size();
            if (n == 1 && s == "?") return "a";
            for (int i = 0; i < n; i++) {
                if (s[i] == '?') {
                    for (char c = 'a'; c <= 'z'; c++) {
                        if ((i > 0 && s[i-1] == c) || (i < n-1 && s[i+1] == c)) continue;
                        s[i] = c;
                        break;
                    }
                }
            }
            return s;
        }
    };